vm/src/object/core.rs

```diff
@@ -711,14 +712,37 @@ impl PyObject {
 
     /// Set the dict field. Returns `Err(dict)` if this object does not have a dict field
     /// in the first place.
-    pub fn set_dict(&self, dict: PyDictRef) -> Result<(), PyDictRef> {
-        match self.instance_dict() {
-            Some(d) => {
+    pub fn set_dict(&self, dict: PySetterValue<PyDictRef>) -> Option<()> {
+        // NOTE(hanif) - So far, this is the only error condition that I know of so we can use Option
```

Comment:
```suggestion
        // NOTE: So far, this is the only error condition that I know of so we can use Option
```
Leaving name in comment is usually not very helpful than git log or git blame. A common scenario with this comment is that you get questions about this code long time later when the code is totally changed.